### PR TITLE
add `radio-group` test case

### LINKED ISSUE #1459: Enhance Test Coverage for `@yamada-ui/radio`
package: @yamada-ui/radio

### Description
`@yamada-ui/radio` has a test coverage of less than 90%. We will enhance it to above 90%.

### Target files
- [packages/components/radio/src/radio-group.tsx](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/radio/src/radio-group.tsx)
  - [L86](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/radio/src/radio-group.tsx#L86)
  - [L127](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/radio/src/radio-group.tsx#L127)
  - [L217](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/radio/src/radio-group.tsx#L217)

### Problem Statement/Justification
Testing is essential to ensure quality when operating a package.

### Additional Information
While it's ideal to cover all tests, if there are too many lines to work on, please create a PR within your capacity.
Any deficiencies not covered by the PR will be automatically created as issues.

### PR DESCRIPTION
**Pull request by cidkumagai**
Closes #1459 

## Description
Add `radio-group` test case

### CODE REVIEW

File: .changeset/slimy-windows-poke.md
Comment (suzukisan22) [inline .changeset/slimy-windows-poke.md:1]: No need changeset

Comment (108yen) [reply .changeset/slimy-windows-poke.md:1]: @suzukisan22 @cidkumagai 
Still under review, but `changeset` may be necessary due to changes in `radio-group.tsx`.

File: packages/components/radio/tests/radio-group.test.tsx [hunk 1]
```diff
@@ -61,4 +64,162 @@ describe("<RadioGroup/>", () => {
...
+    const { result } = renderHook(() => useRadioGroup({ defaultValue: "1" }))
+
+    act(() => {
+      result.current.onChange("1")
+    })
+
+    expect(result.current.value).toBe("1")
+  })
+
+  test("should return checked attribute when isNative is false (with focus)", () => {
```
Comment (108yen) [inline packages/components/radio/tests/radio-group.test.tsx:191]: Is this test correct?

File: packages/components/radio/tests/radio-group.test.tsx [hunk 2]
```diff
@@ -61,4 +64,162 @@ describe("<RadioGroup/>", () => {
...
+    const radioOne = screen.getByTestId("radio-1")
+    const radioTwo = screen.getByTestId("radio-2")
+
+    await user.tab()
+
+    expect(radioOne.querySelector(".ui-radio__input")).not.toHaveFocus()
+    expect(radioTwo.querySelector(".ui-radio__input")).toHaveFocus()
+  })
+
+  test("onChange with focus should move to the second Radio", async () => {
```
Comment (108yen) [inline packages/components/radio/tests/radio-group.test.tsx:126]: The test passes even if some elements are removed.
Is this test correct?

File: packages/components/radio/tests/radio-group.test.tsx [hunk 3]
```diff
@@ -61,4 +64,162 @@ describe("<RadioGroup/>", () => {
...
+      </div>,
+    )
+    const radioGroup = screen.getByRole("group")
+    fireEvent.focus(radioGroup)
+
+    const query = `input:not(:disabled)`
+    expect(container.querySelector(query)).toHaveFocus()
+  })
+
+  test("onChange with focus should move to the first Radio", async () => {
```
Comment (108yen) [inline packages/components/radio/tests/radio-group.test.tsx:145]: The test passes even if some elements are removed.
Is this test correct?

File: packages/components/radio/src/radio-group.tsx
Comment (108yen) [inline packages/components/radio/src/radio-group.tsx:1]: What are the reasons why this change has been made?

File: .changeset/slimy-windows-poke.md
Comment (108yen) [reply .changeset/slimy-windows-poke.md:1]: Still under review but `radio-group.tsx` has changed and may need `changeset`.

File: packages/components/radio/src/radio-group.tsx
Comment (cidkumagai) [reply packages/components/radio/src/radio-group.tsx:1]: I initially wrote expect(props.isChecked).toBeFalsy() but removed it because it was unnecessary.🙇‍♂️

**Comment by cidkumagai** [comment]
@108yen 
I reviewed and revised my initial implementation but still feel uncertain. Could you please provide advice and review it again?🙇‍♂️

**Review by 108yen — APPROVED** [review]
Thank you!

**Review by hirotomoyamada — APPROVED** [review]

**Comment by hirotomoyamada** [comment]
Thanks!
